semgrep-core/src/parsing/Parse_rule.ml

```diff
@@ -43,64 +49,107 @@ module H = Parse_mini_rule
 (* Helpers *)
 (*****************************************************************************)
 
-(* less: could use a hash to accelerate things *)
-let rec find_fields flds xs =
-  match flds with
-  | [] -> ([], xs)
-  | fld :: flds ->
-      let fld_match = List.assoc_opt fld xs in
-      let xs = List.remove_assoc fld xs in
-      let matches, rest = find_fields flds xs in
-      ((fld, fld_match) :: matches, rest)
+type env = { id : string; languages : R.xlang; path : string list }
```

Comment:
Assuming we can always point to concrete line and column, do we need the `path` or is it just to provide yet another way of locating the error?

Reply:
This is for the playground, because with basic mode this is a much easier way to locate the error